app/src/components/file-upload/FileUpload.tsx

```diff
@@ -198,41 +223,56 @@ export const FileUpload: React.FC<IFileUploadProps> = (props) => {
     }
   };
 
-  useEffect(() => {
-    if (!fileToRemove) {
+  const removeFile = (fileName: string) => {
+    // Find index of file to remove
+    const index = files.current.findIndex((item) => item.file.name === fileName);
+
+    if (index === -1) {
       return;
     }
 
-    const removeFile = (fileName: string) => {
-      const index = files.findIndex((item) => item.file.name === fileName);
+    // Update array of files
+    const newFiles = [...files.current];
+    newFiles.splice(index, 1);
```

Comment:
You could clean this up a bit using `Array.filter`:
```javascript
const newFiles = files.current.filter((item) => item.file.name !== fileName)
```
How do we expect this method to handle deletion for two files that have the same name? In its current behaviour, this method will delete the first file that matches on the filename, even if the second file is deleted. If possible, I'd manage deletes using the item's index in the array, rather than by filename.

Reply:
We get the index because we use it to remove items from 2 arrays (the file array here, and the array of file upload components just below)